modules/tinymce/src/core/main/ts/newline/InsertBlock.ts

```diff
@@ -133,11 +133,39 @@ const getEditableRoot = function (dom, node) {
   return parent !== root ? editableRoot : root;
 };
 
+const applyAttributes = (editor: Editor, node: DomElement, forcedRootBlockAttrs: Record<string, string>) => {
+  // Merge and apply style attribute
+  Option.from(forcedRootBlockAttrs.style)
+    .map(editor.dom.parseStyle)
+    .each((attrStyles) => {
+      const currentStyles = Css.getAllRaw(Element.fromDom(node));
+      const newStyles = { ...currentStyles, ...attrStyles };
+      editor.dom.setStyles(node, newStyles);
+    });
+
+  // Merge and apply class attribute
+  Option.from(forcedRootBlockAttrs.class).map((attrClasses) => attrClasses.split(/\s+/)).each((attrClasses) => {
+    Option.from(node.className)
+      .map((currentClasses) => Arr.filter(currentClasses.split(/\s+/), (c) => c !== ''))
+      .each((currentClasses) => {
+        const filteredClasses = Arr.filter(currentClasses, (c) => !Arr.contains(attrClasses, c));
```

Comment:
`Options.lift2` might help here

Do the mapping for `attrClasses` and `currentClasses` into separate variables, then use lift2 in place of the nested `each` calls

Reply:
Using `lift2` is much cleaner. Thanks for the suggestion.